### PR TITLE
Add comprehensive test cases for IPv4 and IPv6 conversions 

### PR DESCRIPTION
**Pull request by vidhvath28**
**Description:** 
### This PR adds extensive test coverage for the IPv4 and IPv6 conversion utilities. It includes:

1. **Boundary Testing**: Added test cases for the boundary values in both IPv4 (0.0.0.0, 255.255.255.255) and IPv6 (::, ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff).
2. **Arithmetic Testing:** Introduced test cases to check the correct behavior of arithmetic operations on IP addresses.
3. **Error Handling:** Validated the proper error handling for invalid inputs such as incorrect IP address strings and out-of-range values.
4. **Edge Cases:** Covered scenarios with leading zeros, the smallest and largest possible IP addresses, negative numbers, and mixed formats (IPv4 and IPv6).
5. **Consistency Checks:** Ensured conversions between strings and integers/BigInts for both IPv4 and IPv6 are consistent and reversible.

_The added test cases should improve the robustness of the IP conversion utilities and ensure that edge cases and potential bugs are caught early._

### CODE REVIEW

**Comment by 0xflotus** [comment]
Thank you for your contribution. 
Can you please check the [errors in the test step of the actions](https://github.com/0xflotus/ip2d/actions/runs/11138290804/job/30953057470?pr=680) and fix this? If tests are failing which failed before your PR, it's ok, but it should successful compile.

**Comment by vidhvath28** [comment]
> Thank you for your contribution. Can you please check the [errors in the test step of the actions](https://github.com/0xflotus/ip2d/actions/runs/11138290804/job/30953057470?pr=680) and fix this? If tests are failing which failed before your PR, it's ok, but it should successful compile.



> Thank you for your contribution. Can you please check the [errors in the test step of the actions](https://github.com/0xflotus/ip2d/actions/runs/11138290804/job/30953057470?pr=680) and fix this? If tests are failing which failed before your PR, it's ok, but it should successful compile.

I have tried compiling before submitting and did the test now also so, the compilation works successfully

File: index.spec.ts [hunk 1]
```diff
@@ -51,6 +51,45 @@ test("arithmetic example", () => {
...
+});
+
+test("should handle IPv4 with leading zeros", () => {
+  expect(fromIPv4("192.168.001.001")).toBe(fromIPv4("192.168.1.1"));
+  expect(toIPv4(fromIPv4("001.002.003.004"))).toBe("1.2.3.4");
+});
+
+test("should throw error for numbers beyond IPv4 range", () => {
+  expect(() => toIPv4(4294967296)).toThrow(); // beyond 32-bit range
+  expect(() => toIPv4(BigInt(4294967296))).toThrow();
```
Comment (0xflotus) [inline index.spec.ts:80]: This line is the problem. You give `toIPv4()` a `bigint` but `number` is expected. Like in the line above. If you delete this line, it will work.

**Review by 0xflotus — CHANGES_REQUESTED** [review]

File: index.spec.ts (hunk 1 above)
Comment (vidhvath28) [reply index.spec.ts]: I removed those lines from the test file has mentioned .

**Review by 0xflotus — APPROVED** [review]

**Comment by 0xflotus** [comment]
Some tests are failing. Let's look at this in another issue.
